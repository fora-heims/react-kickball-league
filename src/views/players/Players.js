import { React, useEffect, useState } from 'react';
import './Players.css';
import PlayerList from '../../components/playerlist/PlayerList.js';
import { getPlayers } from '../../services/players';

export default function Players() {
  const [players, setPlayers] = useState([]);

  useEffect(() => {
    const fetchData = async () => {
      const playersData = await getPlayers();
      setPlayers(playersData);
    };
    fetchData();
  }, []);

  return (
    <div>
      <h1>Players</h1>
      <PlayerList players={players}></PlayerList>
    </div>
  );
}
