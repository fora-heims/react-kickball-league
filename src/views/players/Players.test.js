import { render } from '@testing-library/react';
import Players from './Players.js';

test('', () => {
  const container = render(<Players />);
  expect(container).toMatchSnapshot();
});
