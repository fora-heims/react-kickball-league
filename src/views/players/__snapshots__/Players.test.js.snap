// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h1>
          Players
        </h1>
        <div>
          <label>
            Team:
            <select>
              <option
                value="all"
              >
                All Teams
              </option>
              <option
                value="1"
              >
                Bridge City Sneakers
              </option>
              <option
                value="2"
              >
                Stumptown Lumberjacks
              </option>
              <option
                value="3"
              >
                Lakeville Thunderfeet
              </option>
              <option
                value="5"
              >
                Vancougers
              </option>
              <option
                value="4"
              >
                Mt. Hoodies
              </option>
            </select>
          </label>
          <ul />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h1>
        Players
      </h1>
      <div>
        <label>
          Team:
          <select>
            <option
              value="all"
            >
              All Teams
            </option>
            <option
              value="1"
            >
              Bridge City Sneakers
            </option>
            <option
              value="2"
            >
              Stumptown Lumberjacks
            </option>
            <option
              value="3"
            >
              Lakeville Thunderfeet
            </option>
            <option
              value="5"
            >
              Vancougers
            </option>
            <option
              value="4"
            >
              Mt. Hoodies
            </option>
          </select>
        </label>
        <ul />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
